ddtrace/internal/compat.py

```diff
@@ -84,6 +85,27 @@
 else:
     pattern_type = re._pattern_type  # type: ignore[misc,attr-defined]
 
+try:
+    from inspect import getargspec as getfullargspec
+
+    def args_provided(f, argspec):
```

Comment:
Whilst the logic is correct, I think `args_provided` is a misnomer for this helper as `isgeneratorfunction` is not a check on arguments. IMO something more appropriate is `is_argless_function`/`is_void_function` ... .